### DISCUSSION TITLE
Change non-existant short URL favicon without modifying core files

### DISCUSSION THREAD

**Discussion by jeffshead** [YOURLS/YOURLS#3473] [Q&A]
I know about and I have added my favicon to the /user directory. That works but I want to change the chain-link favicon that is displayed when a non-existant short URL is used by a visitor.

Below is the relevant code that is responsible for displaying a favicon:
**yourls-loader.php (lines 2-8):**
```
// Handle inexistent root favicon requests and exit
if ( '/favicon.ico' == $_SERVER['REQUEST_URI'] ) {
	header( 'Content-Type: image/gif' );
	echo base64_decode( "R0lGODlhEAAQAJECAAAAzFZWzP///wAAACH5BAEAAAIALAAAAAAQABAAAAIplI+py+0PUQAgSGoNQFt0LWTVOE6GuX1H6onTVHaW2tEHnJ1YxPc+UwAAOw==" );
	exit;
}
```

I'm trying to avoid core file mods. Is there a hook or a constant that can be used to replace this code so that I can specify a custom favicon or use the same favicon that I have in the /user folder?

**Comment by ozh** [comment]
The thing here is that the favicon trick has to fire before YOURLS is loaded (to lighten things) - so no YOURLS stuff (actions, filters) are available at this time

Maybe just symlink a root /favicon.ico to the one in user/ ?
